### PR TITLE
Update CocoaPods recommended version to 1.9

### LINKED ISSUE #69807: Update CocoaPods recommended version to 1.9
[CocoaPods 1.9](https://blog.cocoapods.org/CocoaPods-1.9.0-beta/) introduced support for xcframeworks.  It was released in [February 2020](https://github.com/CocoaPods/CocoaPods/releases/tag/1.9.0).

Flutter will likely need to start being [distributed as an xcframework](https://github.com/flutter/flutter/issues/60109) to support [Apple Silicon simulators](https://github.com/flutter/flutter/issues/64502).

Blocked by https://github.com/flutter/flutter/issues/69808 and https://github.com/flutter/flutter/issues/69829

### LINKED ISSUE #69808: Upgrade CocoaPods gem version on builders
The version of CocoaPods on the mac builders is 1.8.4
```
Fetching cocoapods 1.8.4
Installing cocoapods 1.8.4
```
https://logs.chromium.org/logs/flutter/buildbucket/cr-buildbucket.appspot.com/8864533259479547856/+/steps/Install_gems/0/steps/install_gems/0/stdout

Upgrade to version 1.9.3.

The tool will soon recommend a minimum of 1.9 https://github.com/flutter/flutter/issues/69807

### PR DESCRIPTION
**Pull request by jmagman**
## Description

[CocoaPods 1.9](https://blog.cocoapods.org/CocoaPods-1.9.0-beta/) introduced support for xcframeworks.  It was released in [February 2020](https://github.com/CocoaPods/CocoaPods/releases/tag/1.9.0).

Flutter will likely need to start being [distributed as an xcframework](https://github.com/flutter/flutter/issues/60109) to support [Apple Silicon simulators](https://github.com/flutter/flutter/issues/64502).  Nudge people to start upgrading.

Screen Shot 2020-11-04 at 12 51 54 PM

## Related Issues

Fixes https://github.com/flutter/flutter/issues/69808
Fixes https://github.com/flutter/flutter/issues/69807

See https://github.com/flutter/flutter/pull/52071 for the last time this was done.

https://github.com/flutter/flutter/issues/60109
https://github.com/flutter/flutter/issues/64502

## Tests

Updated cocoapods_test

### CODE REVIEW

**Comment by jmagman** [comment]
This seems to have worked on the builder:
```
Fetching cocoapods 1.9.3
Installing cocoapods 1.9.3
```
https://logs.chromium.org/logs/flutter/buildbucket/cr-buildbucket.appspot.com/8864525671995568160/+/steps/Install_gems/0/steps/install_gems/0/stdout

**Comment by jmagman** [comment]
This should probably wait for https://github.com/flutter/flutter/issues/69829 to make sure the devicelab machine have a new enough version of CocoaPods but I'm tempted to yolo merge this since it shouldn't actually cause any test failures, just add the doctor warning.

**Review by jonahwilliams — APPROVED** [review]
Tool changes LGTM.

I think the devicelab will not fail if doctor checks don't all come back positive

File: packages/flutter_tools/test/general.shard/macos/cocoapods_test.dart [hunk 1]
```diff
@@ -181,13 +181,13 @@ void main() {
...
       pretendPodIsInstalled();
-      pretendPodVersionIs('1.8.0');
+      pretendPodVersionIs('1.9.0');
       expect(await cocoaPodsUnderTest.evaluateCocoaPodsInstallation, CocoaPodsStatus.recommended);
     });
 
     testWithoutContext('detects above recommended version', () async {
       pretendPodIsInstalled();
-      pretendPodVersionIs('1.8.1');
+      pretendPodVersionIs('1.9.1');
```
Comment (christopherfujino) [inline packages/flutter_tools/test/general.shard/macos/cocoapods_test.dart:190]: not a blocker, but should we be warning if the .z is above the recommended version?

Comment (jmagman) [reply packages/flutter_tools/test/general.shard/macos/cocoapods_test.dart:190]: Hm, what do you mean? This test shows it returning `CocoaPodsStatus.recommended` which doesn't translate to a warning.

Comment (christopherfujino) [reply packages/flutter_tools/test/general.shard/macos/cocoapods_test.dart:190]: ahh, sorry, you're right.

**Review by christopherfujino — APPROVED** [review]
as keeper of the gemfile.lock file, i approve of this message.

**Comment by jmagman** [comment]
Google tests unrelated https://github.com/flutter/flutter/issues/69923

**Pull request merged**